binarySearchPC2.java

```diff
@@ -0,0 +1,46 @@
+//Precourse 2: Exercise 2    --  Implementing Singly Linked List 
+//1441. Build an Array With Stack Operations
+// Time Complexity : O(log n) because we divided the array into half and kept searching in it. In short we straight away reducing the half part of array
+// Space Complexity : O(n) required for storing n elements

```

Comment:
No, Space is constant as you are using any additional memory for searching an element.